### ISSUE TITLE
build mpv 0.37.0 tvOS error

### ISSUE THREAD

**Issue by kingslay** [mpv-player/mpv#12987]
../../../../mpv-v0.37.0/osdep/subprocess-posix.c:53:16: error: 'execve' is unavailable: not available on tvOS
        return execve(file, argv, envp);
               ^
/Applications/Xcode15.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator17.0.sdk/usr/include/unistd.h:445:6: note: 'execve' has been explicitly marked unavailable here
int      execve(const char * __file, char * const * __argv, char * const * __envp) __WATCHOS_PROHIBITED __TVOS_PROHIBITED;
         ^
../../../../mpv-v0.37.0/osdep/subprocess-posix.c:64:13: error: 'execve' is unavailable: not available on tvOS
            execve(fn, argv, envp);
            ^
/Applications/Xcode15.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator17.0.sdk/usr/include/unistd.h:445:6: note: 'execve' has been explicitly marked unavailable here
int      execve(const char * __file, char * const * __argv, char * const * __envp) __WATCHOS_PROHIBITED __TVOS_PROHIBITED;
         ^
../../../../mpv-v0.37.0/osdep/subprocess-posix.c:111:12: error: 'fork' is unavailable: not available on tvOS
    fres = fork();
           ^
/Applications/Xcode15.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator17.0.sdk/usr/include/unistd.h:447:8: note: 'fork' has been explicitly marked unavailable here
pid_t    fork(void) __WATCHOS_PROHIBITED __TVOS_PROHIBITED;
         ^
../../../../mpv-v0.37.0/osdep/subprocess-posix.c:226:22: error: 'fork' is unavailable: not available on tvOS
        pid_t fres = fork();
                     ^
/Applications/Xcode15.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator17.0.sdk/usr/include/unistd.h:447:8: note: 'fork' has been explicitly marked unavailable here
pid_t    fork(void) __WATCHOS_PROHIBITED __TVOS_PROHIBITED;
         ^
4 errors generated.

**Comment by andrefmsilva** [comment]
[tvos.patch](https://github.com/mpv-player/mpv/files/13486594/tvos.patch)
@kingslay try this one. You need to enable the meson tvos option `-Dtvos=true`

**Comment by andrefmsilva** [comment]
@kingslay I don't remember if you need this one too:
[patchs0001.patch](https://github.com/mpv-player/mpv/files/13486682/patchs0001.patch)

also you should change your repo to use meson for MPV. Thanks for really nice player!

**Comment by andrefmsilva** [comment]
@Dudemanguy  @sfan5  I don't know if the first patch can be done automatically. I've been using it to compile latest mpv versions for tvOS.

**Comment by kingslay** [comment]
Yes, I am currently using subprocess-dummy.c to temporarily solve the tvos problem. But hopefully mpv can provide a better solution. Because I don’t know if there will be any impact if the mp_subprocess2 function is not implemented. @andrefmsilva   @Dudemanguy

**Comment by Dudemanguy** [comment]
Not having subprocesses implemented means that scripting and such will not work. Short of implementing your own `fork` and `execve` for tvOS, there's not really anything we can do about that. As far as patches go, instead of a build option I would suggest checking for the existence of fork and using `subprocess-dummy.c` if it is not there.

**Comment by karelrooted** [comment]
[mpv-tvos-fix-subprocess.patch](https://github.com/mpv-player/mpv/files/13536239/mpv-tvos-fix-subprocess.patch)

this tvOS patch use subprocess-dummy.c when tvOS is detected

**Comment by Dudemanguy** [comment]
One small nit with your patch (`if host_machine.system() == 'darwin' is unnecessary) but otherwise it looks OK. Could you open up a PR?

**Comment by karelrooted** [comment]
> One small nit with your patch (`if host_machine.system() == 'darwin' is unnecessary`) but otherwise it looks OK. Could you open up a PR?

#13206 use have_func check  to cover more posix system that doesn't have fork func
